pkg/registry/helm.go

```diff
@@ -14,12 +14,14 @@ type HelmChart struct {
 	ContainerImages []string
 }
 
-func HelmCharts(helmCharts []image.HelmChart, valuesDir, buildDir, kubeVersion string, helm image.Helm) ([]*HelmChart, error) {
+func HelmCharts(helm *image.Helm, valuesDir, buildDir, kubeVersion string, helmClient image.HelmClient) ([]*HelmChart, error) {
 	var charts []*HelmChart
+	chartRepoMap := mapChartRepos(helm)
 
-	for _, helmChart := range helmCharts {
+	for _, helmChart := range helm.Charts {
 		c := helmChart
-		chart, err := handleChart(&c, valuesDir, buildDir, kubeVersion, helm)
+		r := chartRepoMap[c.RepositoryName]
```

Comment:
We must ensure that a value is there:
```suggestion
		r, ok := chartRepoMap[c.RepositoryName]
		if !ok {
			return nil, fmt.Errorf("repository not found for chart %s", c.Name)
		}
```